tests/conftest.py

```diff
@@ -25,3 +32,165 @@ def model(monkeypatch) -> capellambse.MelodyModel:
     """Return test model."""
     monkeypatch.setattr(sys, "stderr", io.StringIO)
     return capellambse.MelodyModel(TEST_ROOT / TEST_MODEL)
+
+
+def remove_ids_from_elk_layout(
+    elk_layout: _elkjs.ELKOutputData,
+) -> dict[str, t.Any]:
+    """Remove ids from a layout from ELK."""
+    layout_dict = elk_layout.model_dump(exclude_defaults=True)
+
+    def remove_ids(obj: t.Any) -> t.Any:
+        if isinstance(obj, dict):
+            return {k: remove_ids(v) for k, v in obj.items() if k != "id"}
+        if isinstance(obj, list):
+            return [remove_ids(item) for item in obj]
+        return obj
+
+    return remove_ids(layout_dict)
+
+
+def text_size_mocker(
+    text: str, fonttype: str = "", size: int = 12
+) -> tuple[int, int]:
+    """Mock text size calculation."""
+    del fonttype
+    return (len(text) * size // 2, size)
+
+
+def write_test_data_file(
+    file_path: pathlib.Path,
+    data: _elkjs.BaseELKModel | list[_elkjs.BaseELKModel],
+):
+    """Write test data to file.
+
+    Note
+    ----
+    This is a helper function to write test data to files in case the
+    expected test data changed.
+    """
+    import json
+
+    if isinstance(data, list):
+        data_dump = {
+            "edges": [edge.model_dump(exclude_defaults=True) for edge in data]
+        }
+    else:
+        data_dump = data.model_dump(exclude_defaults=True)
+
+    file_path.write_text(json.dumps(data_dump, indent=4), encoding="utf8")
+
+
+def generic_collecting_test(
+    model: capellambse.MelodyModel,
+    params: tuple[str, str, dict[str, t.Any]],
+    data_root: pathlib.Path,
+    diagram_attr: str,
+    extra_suffix: str | None = None,
+    extra_assert: cabc.Callable[
+        [str, list[_elkjs.ELKInputEdge] | _elkjs.ELKOutputData], bool
+    ]
+    | None = None,
+):
+    """Test collecting test.
+
+    Parameters
+    ----------
+    model
+        The MelodyModel instance.
+    params
+        Tuple consisting of (UUID, file_name, render_params).
+    data_root
+        Path to the JSON input files.
+    diagram_attr
+        The attribute name of the diagram.
+    extra_suffix
+        Optional suffix for an extra file.
+    extra_assert
+        Optional function to assert a statement with an extra file.
+    """
+    uuid, file_name, _ = params
+    obj = model.by_uuid(uuid)
+    file_path = data_root / file_name
+    data_text = file_path.read_text(encoding="utf8")
+    expected_main = _elkjs.ELKInputData.model_validate_json(data_text)
+    expected_extra_file: pathlib.Path | None = None
+    if extra_suffix:
+        expected_extra_file = data_root / (file_path.stem + extra_suffix)
+        expected_extra = expected_extra_file.read_text(encoding="utf8")
+
+    with mock.patch("capellambse.helpers.extent_func") as mock_ext:
+        mock_ext.side_effect = text_size_mocker
+        result = getattr(obj, diagram_attr).elk_input_data({})
```

Comment:
Since we don't care about when and how the mocked function gets called, we can flatten out this block by annotating `generic_collecting_test` with:

```python
@mock.patch("capellambse.helpers.extent_func", text_size_mocker)
```